libvirt/server/service.sls

```diff
@@ -43,10 +43,22 @@ libvirt_sysconfig:
 /etc/default/libvirt-bin:
   file.managed:
   - source: salt://libvirt/files/libvirt-bin
+  - template: jinja
   - require:
     - pkg: libvirt_packages
   - watch_in:
     - service: libvirt_service
+{%- if grains.get('init', None) == 'systemd' %}
+
+libvirt_restart_systemd:
```

Comment:
Are you sure this is needed? You are not changing systemd unit file but only it's config.